<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SerialMain</class>
 <widget class="QMainWindow" name="SerialMain">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>548</width>
    <height>384</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string/>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <layout class="QVBoxLayout" name="verticalLayout" stretch="0,0,0,0,0,0,0">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout" stretch="1,1,0,1,2">
        <item>
         <widget class="QComboBox" name="cbDevice">
          <property name="editable">
           <bool>true</bool>
          </property>
          <item>
           <property name="text">
            <string>Device ???</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>COM1</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>COM2</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>COM3</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>COM4</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>/dev/ttyUSB0</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>/dev/rfcomm0</string>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="sbBaudrate">
          <property name="minimum">
           <number>2400</number>
          </property>
          <property name="maximum">
           <number>921600</number>
          </property>
          <property name="singleStep">
           <number>100</number>
          </property>
          <property name="value">
           <number>115200</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Baud</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btnConnect">
          <property name="text">
           <string>Connect</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="lblDevice">
          <property name="font">
           <font>
            <pointsize>10</pointsize>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>Port ??? </string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QPlainTextEdit" name="edReceive">
        <property name="font">
         <font>
          <family>Courier</family>
         </font>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
        <property name="plainText">
         <string/>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4" stretch="0,1">
        <item>
         <widget class="QCheckBox" name="cbHEXrec">
          <property name="text">
           <string>HEX</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btnClear">
          <property name="text">
           <string>Clear RX</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QCheckBox" name="cbHEX_2">
          <property name="text">
           <string>HEX</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="leToSend_2">
          <property name="text">
           <string>AA 10 10 AB</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btnSend_2">
          <property name="text">
           <string>Send - 2</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="0,1,0">
        <item>
         <widget class="QCheckBox" name="cbHEX_1">
          <property name="text">
           <string>HEX</string>
          </property>
          <property name="checked">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="leToSend_1">
          <property name="text">
           <string>Hello World ;-)</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btnSend_1">
          <property name="text">
           <string>Send - 1</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QPushButton" name="btnSend_File">
        <property name="text">
         <string>Send File</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_6">
        <item>
         <widget class="QCheckBox" name="cbHEX_prot">
          <property name="text">
           <string>HEX</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="leToSend_prot">
          <property name="text">
           <string>70 06 03 08 23 99 99 00 55 55 55 ff</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btnSend_Prot">
          <property name="cursor">
           <cursorShape>ForbiddenCursor</cursorShape>
          </property>
          <property name="toolTip">
           <string>adds STARTFLAG (AA) [ESCFLAGS (AF)] CHKSUM and STOPFLAG (AB)</string>
          </property>
          <property name="text">
           <string>HSUprot</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
